prototypes/ScheduledMerges.hs

```diff
@@ -468,21 +488,32 @@ supply (LSMHandle scr lsmr) credits = do
     supplyCredits credits ls
     invariant ls
 
+data LookupResult v b =
+    NotFound
+  | Found !v !(Maybe b)
+  deriving stock (Eq, Show)
+
 lookups :: LSM s -> [Key] -> ST s [(Key, LookupResult Value Blob)]
-lookups lsm = mapM (\k -> (k,) <$> lookup lsm k)
+lookups lsm ks = do
+    runs <- concat <$> allLayers lsm
+    return $ map (\k -> (k, doLookup k runs)) ks
```

Comment:
Isn't the code before equivalent to the code after?

Reply:
Yes, I just wanted to extract a pure function in preparation for lookups in table unions.